apps/lib/app_provider.c

```diff
@@ -50,13 +90,28 @@ void app_providers_cleanup(void)
 {
     sk_OSSL_PROVIDER_pop_free(app_providers, provider_free);
     app_providers = NULL;
+    OPENSSL_CTX_free(libctx);
+    libctx = NULL;
```

Comment:
If you call `app_new_libctx()` from `main()`, these two lines should move there as well, of course...

Reply:
see above

Reply:
Fixed.